kpatch-build/kpatch-build

```diff
@@ -693,14 +723,23 @@ if [[ $DEBUG -ge 4 ]]; then
 	export KPATCH_GCC_DEBUG=1
 fi
 
-echo "Building original kernel"
-./scripts/setlocalversion --save-scmversion || die
-make mrproper 2>&1 | logger || die
-cp -f "$CONFIGFILE" "$SRCDIR/.config"
-unset KPATCH_GCC_TEMPDIR
-# $TARGETS used as list, no quotes.
-# shellcheck disable=SC2086
-CROSS_COMPILE="$TOOLSDIR/kpatch-gcc " make "-j$CPUS" $TARGETS 2>&1 | logger || die
+if [[ -z "$RESTOREARCHIVE" ]]; then
+	echo "Building original kernel"
+	./scripts/setlocalversion --save-scmversion || die
+	make mrproper 2>&1 | logger || die
+	cp -f "$CONFIGFILE" "$SRCDIR/.config"
+	unset KPATCH_GCC_TEMPDIR
+	# $TARGETS used as list, no quotes.
+	# shellcheck disable=SC2086
+	CROSS_COMPILE="$TOOLSDIR/kpatch-gcc " make "-j$CPUS" $TARGETS 2>&1 | logger || die
```

Comment:
I think we'll either have to use `O=` here or tar $USERSRCDIR for this to work properly with `-s`. If `O=` approach is taken (which I think is a more universal one) actual make command should probably be executed even if we are restoring just to make sure source haven't changed. I have a specific usecase in mind: I build a lot against linus' tree which is constantly updated and with described approach it will be possible to partially use the cache even though the kernel version is different. Because of this case I also don't think that `--restore-cache` and `--save-cache` should be mutually exclusive. I can take on that in a later pullrequest though if you prefer it this way.

Reply:
@sm00th Would removing the "make mrproper" solve your problem? I don't see why the mrproper is needed, and it would be very useful if somebody figured that out, as that seems to be one of the main motivations for this feature.

Also have you tried ccache, or does that not help for some reason?

Reply:
BTW, the use of `O=` is problematic, see 2e99d6b7a44876a4368e7cf7ef6e75f203728723.